### ISSUE TITLE
Skip cuda kernel launch with torch.sum when dimension length is 0

### ISSUE THREAD

**Issue by mvpatel2000** [pytorch/pytorch#110116]
### 🚀 The feature, motivation and pitch

If there is a torch tensor, e.g. `a = torch.ones([1, 1000])` and I call `a.sum(dim=0)`, torch will launch a cuda kernel for this when it's really just a view. Torch should special case this or include it inside the compiler in `torch.compile` as it is leads to significant wins in certain settings.
